libhsh/include/hsh/bits/vulkan_impl.h

```diff
@@ -1057,13 +1061,14 @@ RenderTextureAllocation::RenderTextureAllocation(
   assert(Surface);
   assert(NumColorBindings <= MaxRenderTextureBindings);
   assert(NumDepthBindings <= MaxRenderTextureBindings);
+  PreRender();
```

Comment:
This part is a little odd, you may have a better idea of how to accomplish this.

Currently this is necessary so that `render_texture2d` can be bound during the same render pass that it's created.